mars/deploy/local/session.py

```diff
@@ -148,26 +150,34 @@ def run(self, *tileables, **kw):
         else:
             return self.fetch(*tileables)
 
-    def fetch(self, *tileables):
+    def fetch(self, *tileables, **kw):
         from ...tensor.indexing import TensorIndex
         from ...dataframe.indexing import DataFrameIlocGetItem
 
+        fetch_indexes = kw.pop('indexes', ())
+
         tileable_results = []
         for tileable in tileables:
             # TODO: support DataFrame getitem
-            if tileable.key not in self._executed_tileables and \
-                    isinstance(tileable.op, (TensorIndex, DataFrameIlocGetItem)):
-                key = tileable.inputs[0].key
-                indexes = tileable.op.indexes
-                if not all(isinstance(ind, (slice, Integral)) for ind in indexes):
-                    raise ValueError('Only support fetch data slices')
+            if isinstance(tileable, six.string_types):
```

Comment:
Is it possible to save the executed tileable objects on the cluster side, so that we can check here if the tileable object is executer or not. `session.fetch(tileable.key)` looks unpleasant indeed. 

What's your idea about this? @hekaisheng